clients/js/lib/serializer/json_serializer.js

```diff
@@ -1,5 +1,8 @@
 export function createJsonSerializer() {
     return function serialize(data) {
+        if (!data) {
+            throw new Error("Invalid data object for serialization");
+        }
 
         const jsonString = JSON.stringify(data);
```

Comment:
Any reason we are returning string here and not a byte array since, it's going to be exclusively used for event_bytes field.

Reply:
we are returning byte array only, if you will check line number 9